pkg/backend/diy/backend.go

```diff
@@ -775,7 +775,18 @@ func (b *diyBackend) ListStacks(
 
 		chk, err := b.getCheckpoint(ctx, stackRef)
 		if err != nil {
-			return nil, nil, err
+			// There is a race between listing stacks and getting their checkpoints.  If there's an error getting
+			// the checkpoint, check if the stack still exists before returning an error.
+			stacks, stackErr := b.getStacks(ctx)
```

Comment:
Use stackExists rather than listing the whole bucket again.

Reply:
Ah neat that's much better!